src/components/ListItem.tsx

```diff
@@ -83,6 +84,30 @@ export function ListItemCheckBox({ item, listPath }: Props) {
 		}
 	};
 
+	const editItemQuantity = async (quantity: number) => {
+		console.log("Item quantity edited:", quantity);
+
+		if (!listPath) {
+			toast.error("Error: listPath is missing or invalid.");
+			return;
+		}
+
+		if (quantity === 0) {
+			deleteItemHandler();
+		}
```

Comment:
Hmm I'm thinking it may be a smoother user experience if we don't give them the option to put 0 and just regularly delete the item with the delete button. Let me know if folks think otherwise, but I do appreciate the thoughtfulness of it.

Reply:
Thank you for noticing, and thank you for the feedback!

WIP: I removed the deleteItemHandler call when the quantity is equal to "0". I added a "min" attribute to ItemQuantityForm's input and set it to 1.

![image](https://github.com/user-attachments/assets/9a0ebeab-f85d-4e49-b73f-2279675b8d70)
![image](https://github.com/user-attachments/assets/cd1429ac-b18e-4dea-aa99-434fde569220)

Reply:
I basically went back to my original premise: I want to keep a "separation of concerns". Sure, re-usable components are cool too. 

SO
I made the item quantity just another input on the Add Item Form.
![image](https://github.com/user-attachments/assets/efb421e5-017f-488c-9680-0c8caab727fa)
This resolved A LOT of bugs, including features that weren't happening that were supposed to be happening with the attribute being set to "number". 
I also took the time to remove the "How many?" label from the input of the InputQualityForm component.
![image](https://github.com/user-attachments/assets/5ec91f5c-0aff-48b7-af4c-90ebca4b0972)